### DISCUSSION TITLE
"no supported bridge for token pair" - while bridging a custom token

### DISCUSSION THREAD

**Discussion by rube-de** [ethereum-optimism/developers#300] [Application Development]
### Did you check the documentation?

- [X] Yes

### Did you check for duplicate questions?

- [X] Yes

### Issue Description

I deployed an ERC20 token OP Mainnet, that implements the `IOptimismMintableERC20` according to  https://docs.optimism.io/builders/dapp-developers/tutorials/standard-bridge-custom-token
just adding the Permit Extension and changing the `burn` function to the same as in the standard `OptimismMintableERC20`

I used as bridge address in the constructor '0x4200000000000000000000000000000000000007' as in the docs 
(I tried it additionally with '0x4200000000000000000000000000000000000010' what seems the Factory for standard tokens is using)

Using the following L1 and L2 contracts
L1 contract: https://etherscan.io/address/0xB58E61C3098d85632Df34EecfB899A1Ed80921cB
L2 contract: https://optimistic.etherscan.io/address/0x1e82e445f4069fc03ba4cb6de1490b29ac119772

Bridging with the Optimism SDK I get this error:
[`no supported bridge for token pair`](https://github.com/ethereum-optimism/optimism/blob/f3d0102adddce9d33a9daa81fb3195c7a30890b8/packages/sdk/src/cross-chain-messenger.ts#L532)

Is there anything additional needed that the cross-chain-messenger recognises the pair? 

### Additional Information

Before I deployed the L2 custom token, there was already a L2 standard token for the same L1 deployed. For the standard token the bridging works with the Optimism SDK. 
Is there some interference when two L2 tokens are deployed for the same L1 token? As I set both, the L1 and L2 token when bridging this shouldn't be, no?

### Feedback

The docs say for the bridge address is
`_BRIDGE:      "0x4200000000000000000000000000000000000007"`
but the Factory seems to use `0x4200000000000000000000000000000000000010`
is the the docs out of date or correct that for the custom and the standard token there are two different bridge contracts used?

**Comment by sbvegan** [comment]
I noticed this [fix](https://github.com/ethereum-optimism/optimism/blob/a4818e9dcab40d04a7a0e80d424e56f742b384a4/packages/sdk/CHANGELOG.md#318) in the change log. It seems relevant.

What version of the SDK are you using? Can you try again with the latest version?

**Reply by clemsos** [reply]
Hello, I am having the same error - with the latest version of the sdk (3.2.3)

**Reply by clemsos** [reply]
After investigation, the error is in the `docs`. Looking at the sdk, the code will check if `l1Token` and `l2Bridge` are correctly set in the contract, while the variable in the example contract are named differently

https://github.com/ethereum-optimism/optimism/blob/a039f5a7c1fadd680e95d53e315d8fca2cff39a6/packages/sdk/src/adapters/standard-bridge.ts#L173

from https://docs.optimism.io/builders/app-developers/tutorials/standard-bridge-custom-token

```solidity
// @notice Address of the corresponding version of this token on the remote chain.
address public immutable REMOTE_TOKEN;
 
/// @notice Address of the StandardBridge on this network.
address public immutable BRIDGE;
```

Changing the variable names on the l2 contract fixes it.

As an addition to a docs fix, it will be nice to return the actual error if possible.

**Reply by clemsos** [reply]
Also as @rube-de mentioned, the bridge address has to be `0x4200000000000000000000000000000000000010`

**Comment by rube-de** [comment]
> I noticed this [fix](https://github.com/ethereum-optimism/optimism/blob/a4818e9dcab40d04a7a0e80d424e56f742b384a4/packages/sdk/CHANGELOG.md#318) in the change log. It seems relevant.
> 
> What version of the SDK are you using? Can you try again with the latest version?

Using 3.2.0

I now integrated the legacy interface  `ILegacyMintableERC20` which also supports getting the L1 token via `l1Token()` instead of the `remoteToken()` which is in the `IOptimismMintableERC20` interface (the one which is in the example code of the custom-token tutorial)

With this legacy interface the bridging works with the SDK

The legacy call is still even in the developer branch see:
https://github.com/ethereum-optimism/optimism/blob/develop/packages/sdk/src/adapters/standard-bridge.ts#L173

The legacy call is fine, but the SDK should also support the new interface.
